src/visitpy/common/PyText2DObject.C

```diff
@@ -566,123 +566,81 @@ Text2DObject_setattr(PyObject *self, char *name, PyObject *args)
-    if(obj->data->GetFontShadow())
-        fprintf(fp, "fontShadow = 1\n");
-    else
-        fprintf(fp, "fontShadow = 0\n");
-
-    return 0;
-}
-
-static PyObject *
-PyText2DObject_StringRepresentation(const AnnotationObject *atts)
+std::string
+PyText2DObject_ToString(const AnnotationObject *atts, const char *prefix)
 {
     std::string str;
     char tmpStr[1000];
 
     if(atts->GetVisible())
-        snprintf(tmpStr, 1000, "visible = 1\n");
+        snprintf(tmpStr, 1000, "%svisible = 1\n", prefix);
     else
-        snprintf(tmpStr, 1000, "visible = 0\n");
+        snprintf(tmpStr, 1000, "%svisible = 0\n", prefix);
     str += tmpStr;
     if(atts->GetActive())
-        snprintf(tmpStr, 1000, "active = 1\n");
+        snprintf(tmpStr, 1000, "%sactive = 1\n", prefix);
     else
-        snprintf(tmpStr, 1000, "active = 0\n");
-/*CUSTOM*/
-    {   const double *position = atts->GetPosition();
-        snprintf(tmpStr, 1000, "position = (%g, %g)\n", position[0], position[1]);
-    }
+        snprintf(tmpStr, 1000, "%sactive = 0\n", prefix);
+
+    const double *position = atts->GetPosition();
+    snprintf(tmpStr, 1000, "%sposition = (%g, %g)\n", prefix, position[0], position[1]);
+
     str += tmpStr;
     const double *position2 = atts->GetPosition2();
-    snprintf(tmpStr, 1000, "height = %g\n", position2[0]);
+    snprintf(tmpStr, 1000, "%sheight = %g\n", prefix, position2[0]);
     str += tmpStr;
     const unsigned char *textColor = atts->GetTextColor().GetColor();
-    snprintf(tmpStr, 1000, "textColor = (%d, %d, %d, %d)\n", int(textColor[0]), int(textColor[1]), int(textColor[2]), int(textColor[3]));
+    snprintf(tmpStr, 1000, "%stextColor = (%d, %d, %d, %d)\n", prefix, int(textColor[0]), int(textColor[1]), int(textColor[2]), int(textColor[3]));
     str += tmpStr;
     if(atts->GetUseForegroundForTextColor())
-        snprintf(tmpStr, 1000, "useForegroundForTextColor = 1\n");
+        snprintf(tmpStr, 1000, "%suseForegroundForTextColor = 1\n", prefix);
     else
-        snprintf(tmpStr, 1000, "useForegroundForTextColor = 0\n");
+        snprintf(tmpStr, 1000, "%suseForegroundForTextColor = 0\n", prefix);
     str += tmpStr;
     const stringVector &s = atts->GetText();
-    snprintf(tmpStr, 1000, "text = \"%s\"\n", s.size() > 0 ? s[0].c_str() : "");
+    snprintf(tmpStr, 1000, "%stext = \"%s\"\n", prefix, s.size() > 0 ? s[0].c_str() : "");
     str += tmpStr;
     const char *fontFamily_names = "Arial, Courier, Times";
     if(atts->GetFontFamily() == AnnotationObject::Arial)
-        snprintf(tmpStr, 1000, "fontFamily = Arial  # %s\n", fontFamily_names);
+        snprintf(tmpStr, 1000, "%sfontFamily = %sArial  # %s\n", prefix, prefix, fontFamily_names);
     else if(atts->GetFontFamily() == AnnotationObject::Courier)
-        snprintf(tmpStr, 1000, "fontFamily = Courier  # %s\n", fontFamily_names);
+        snprintf(tmpStr, 1000, "%sfontFamily = %sCourier  # %s\n", prefix, prefix, fontFamily_names);
     else
```

Comment:
snprintf is fine but the C++ ways of doing string formatting with string streams are better 👀👀👀👀👀👀

Reply:
not suggesting any changes just wanted to be on my soapbox